modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1289,7 +1289,7 @@ def binary_operation(cls, left, func, right):
                 [
                     part.apply(
                         func,
-                        right[row_idx][col_idx]._data,
+                        right[row_idx][col_idx].list_of_blocks[0],
```

Comment:
Could you add an assertion that `len(list_of_blocks) == 1` for every right partition, and add a TODO linking to #4691? In case there are multiple blocks, I would much rather raise an assertion error here than try to do the multiply with a single block.

Reply:
I agree, continuing of execution just hides problem. The check was added